### ISSUE TITLE
AIRAC: WMS-T layers are not recognized in AIRAC Tool

### ISSUE THREAD

**Issue by kgaeb** [geosolutions-it/austrocontrol-C125#300]
Our (AIRAC) tool for time dependend layers (WMS-T) doesn't recognize WMS-T layers when added to the ToC. In the example below we've added the "Aerodrome" layer manually and the CTR layer without any manual intervention.
Could you please clarifiy, if the tool can/should automatically recognize  a time dependend layer (WMS-T layer)?
What has to be done in the localConfig.json to remove the Timetool in the bottom?
![image](https://user-images.githubusercontent.com/34339627/190570008-93fd941b-f480-4df1-aa0d-e83c132f55a7.png)

**Comment by tdipisa** [comment]
@kgaeb 

>Could you please clarifiy, if the tool can/should automatically recognize a time dependend layer (WMS-T layer)?

Looking at the original issue it seems it should do that. We have to check. If it is a regression we have to check also if that regression was present even before this update. 

>Is there a possibility in the map.json to set the AIRAC tool to the current date?

What is the map.json file? Do you mean maybe the localConfig.json one?

I'm not sure, this is a really hold development. I will ask our developers to check.

>What has to be done in the localConfig.json to remove the Timetool in the bottom?

Remove from configuration [here](https://github.com/geosolutions-it/austrocontrol-C125/blob/master/configs/localConfig.json#L562) _Timeline_ and  _Playback_.

**Comment by kgaeb** [comment]
@tdipisa 



> > Could you please clarifiy, if the tool can/should automatically recognize a time dependend layer (WMS-T layer)?
> 
> Looking at the original issue #3 it seems it should do that. We have to check. If it is a regression we have to check also if that regression was present even before this update.

--> still open



> > Is there a possibility in the map.json to set the AIRAC tool to the current date?
> 
> What is the map.json file? Do you mean maybe the localConfig.json one?

What I meant is the json of the map configuration - stored in the DB and exported via the "Export Map" in the GUI



> > What has to be done in the localConfig.json to remove the Timetool in the bottom?
> 
> Remove from configuration [here](https://github.com/geosolutions-it/austrocontrol-C125/blob/master/configs/localConfig.json#L562) _Timeline_ and _Playback_.

Got that - Thx!

**Comment by dsuren1** [comment]
@kgaeb 
> Could you please clarifiy, if the tool can/should automatically recognize a time dependend layer (WMS-T layer)?

Yes, the tool automatically recognizes the time dependent layer. Kindly refer the gif, where I add a time layer from gs-stable for example, along side `CTR - Control zone` 
**Note**: This was performed in test env https://austrocontrol.geo-solutions.it/austrocontrol-ms2 and here `CTR - Control zone` doesn't have dimension information

![time_layer](https://user-images.githubusercontent.com/26929983/193547964-046ddcc5-5e9a-4e5c-9548-6c70d29b24ec.gif)


> Is there a possibility in the map.json to set the AIRAC tool to the current date?

Unfortunately, no. 
The map.json file is a collection of data to render a map with all the configs preserved when the json is imported back into the application. However, the plugin configuration are done via `localConfig.json`. But as I checked, the plugin doesn't support any date configuration at the moment.

**Comment by kgaeb** [comment]
> > Could you please clarifiy, if the tool can/should automatically recognize a time dependend layer (WMS-T layer)?
> 
> Yes, the tool automatically recognizes the time dependent layer and enables. Kindly refer the gif, where I add a time layer from _gs-stable_ (our demo geoserver) for example, along side `CTR - Control zone` **Note**: This was performed in test env https://austrocontrol.geo-solutions.it/austrocontrol-ms2 and here `CTR - Control zone` doesn't have dimension information

The GetCapabilities request says, that there is a time dependency - the layer is CTR_EFFDATE; Even QGIS recognizes the time component of the layer.

Could you please check again, that you used the correct layer "CTR_EFFDATE", as we have a "CTR" layer as well in the data set (which is not time dimensioned)?

The request to the GetCapabilities is: https://sdigeo-free.austrocontrol.at/geoserver/free/wms?service=WMS&version=1.3.0&request=GetCapabilities

The part of the GetCapabilities for the layer CTR_EFFDATE is:
_

```
> <Layer queryable="1" opaque="0">
> <Name>CTR_EFFDATE</Name>
> <Title>CTR - Control zone [Effective date enabled]</Title>
> <Abstract>CTR, Control Zone, Austria only, effective date, publication cycles</Abstract>
> <KeywordList>
> <Keyword>features</Keyword>
> <Keyword>CTR_EFFDATE</Keyword>
> </KeywordList>
> <CRS>EPSG:4326</CRS>
> <CRS>CRS:84</CRS>
> <EX_GeographicBoundingBox>
> <westBoundLongitude>11.001666667</westBoundLongitude>
> <eastBoundLongitude>16.840833333</eastBoundLongitude>
> <southBoundLatitude>46.538888889</southBoundLatitude>
> <northBoundLatitude>48.306111111</northBoundLatitude>
> </EX_GeographicBoundingBox>
> <BoundingBox CRS="CRS:84" minx="11.001666667" miny="46.538888889" maxx="16.840833333" maxy="48.306111111"/>
> <BoundingBox CRS="EPSG:4326" minx="46.538888889" miny="11.001666667" maxx="48.306111111" maxy="16.840833333"/>
> <BoundingBox CRS="EPSG:3857" minx="1224699.9312477713" miny="5867131.123299828" maxx="1874712.9911639479" maxy="6157932.748450538"/>
> <BoundingBox CRS="EPSG:3034" minx="2213595.09707496" miny="4071818.5636176215" maxx="2425838.777564348" maxy="4507366.045858862"/>
> <BoundingBox CRS="EPSG:4258" minx="46.538888931649836" miny="11.001666667" maxx="48.30611115300715" maxy="16.840833333"/>
> **<Dimension name="time" default="PRESENT" units="ISO8601">2020-11-06T00:00:00.000Z,2022-01-27T00:00:00.000Z</Dimension>**
> <Style>
> <Name>generic</Name>
> <Title>Generic</Title>
> <Abstract>Generic style</Abstract>
> <LegendURL width="20" height="20">
> <Format>image/png</Format>
> <OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink" xlink:type="simple" xlink:href="https://sdigeo-free.austrocontrol.at/geoserver/free/ows?service=WMS&version=1.3.0&request=GetLegendGraphic&format=image%2Fpng&width=20&height=20&layer=CTR_EFFDATE"/>
> </LegendURL>
> </Style>
> </Layer>
```

_

**Comment by dsuren1** [comment]
@kgaeb 
When adding a time dependant layer, the application dispatches `DescribeDomains` call by updating the layer url to a multidimensional url **ex**: url with `/wms` to `/gwc/service/wmts`.
Dispatching this call fails with **404**, due to this dimension domain information are not obtained. Eventually this causes the time tool to stay disabled
https://sdigeo-free.austrocontrol.at/geoserver/free/gwc/service/wmts?service=WMTS&REQUEST=DescribeDomains&version=1.0.0&layer=CTR_EFFDATE&tileMatrixSet=EPSG:4326&expandLimit=10

![image](https://user-images.githubusercontent.com/26929983/193796050-fe1aee38-ceb9-4a8e-a99d-d9d0fee81687.png)
